cmd/kube-apiserver/app/server.go

```diff
@@ -241,6 +243,25 @@ func Run(s *options.ServerRunOptions) error {
 		}
 	}
 
+	client, err := internalclientset.NewForConfig(genericConfig.LoopbackClientConfig)
+	if err != nil {
+		kubeAPIVersions := os.Getenv("KUBE_API_VERSIONS")
+		if len(kubeAPIVersions) == 0 {
+			return fmt.Errorf("failed to create clientset: %v", err)
+		}
+
+		// KUBE_API_VERSIONS is used in test-update-storage-objects.sh, disabling a number of API
+		// groups. This leads to a nil client above and undefined behaviour further down.
+		// TODO: get rid of KUBE_API_VERSIONS or define sane behaviour if set
+		glog.Errorf("Failed to create clientset with KUBE_API_VERSIONS=%q. KUBE_API_VERSIONS is only for testing. Things will break.", kubeAPIVersions)
+	}
+
+	// NOTE(ericchang): Internal informers should switch to using
+	// 'pkg/client/informers/informers_generated', the second informer
+	// created here.
+	sharedInformers := informers.NewSharedInformerFactory(nil, client, 10*time.Minute)
+	internalSharedInformers := internalversion.NewSharedInformerFactory(client, 10*time.Minute)
```

Comment:
After hunting it down it looks like how often it calls the underlying cache's resync method. Which seems to be this:

```
func (c *threadSafeMap) Resync() error {
	// Nothing to do
	return nil
}
```

https://github.com/kubernetes/client-go/blob/86a2be1b447d7abddae88de0a5f642935992b803/tools/cache/thread_safe_store.go#L277

So, I'm a bit confused.